support for mobile layout

sotiwin opened the pull request:
### Summary 🎯

I added support for mobile layout: on mobile the table of content is displayed in the left sidebar: 

![image](https://github.com/user-attachments/assets/a1ee3398-5d31-464c-9a22-86ec1d4a8430)

One thing is missing, but I couldn't figure it out. After clicking on something the sidebar remains opened. 

Additionally there are a few minor changes:
- to allow development in browser
- there was a react error on the console: each child in a list should have unique key

sotiwin committed:
fix development in browser

sotiwin committed:
fix development in browser

sotiwin committed:
Merge remote-tracking branch 'origin/main'

sotiwin committed:
mobile support: on mobile display TOC widget on the left sidebar

sotiwin committed:
fix "Each child in a list should have a unique "key" prop." error

sotiwin committed:
update react-refresh-webpack-plugin

sotiwin committed:
prepare for unlisted publication

toxoxi commented on src/widgets/index.tsx:
```diff
@@ -7,10 +7,18 @@ import {
 async function onActivate(plugin: ReactRNPlugin) {
   // Register a sidebar widget.
-  await plugin.app.registerWidget('table_of_contents', WidgetLocation.RightSidebar, {
+  let os = await plugin.app.getOperatingSystem();
```
[IMO] using const would be better as long as no override after declaration

toxoxi commented on src/widgets/index.tsx:
```diff
@@ -7,10 +7,18 @@ import {
-  await plugin.app.registerWidget('table_of_contents', WidgetLocation.RightSidebar, {
+  let os = await plugin.app.getOperatingSystem();
+
+  let location = WidgetLocation.RightSidebar;
+  if (isMobileOs(os)) {
+    location = WidgetLocation.LeftSidebar
+  }
```
[IMO] As same as the above, using const would be more robust.
```suggestion
  const location = isMobileOs(os) ? WidgetLocation.LeftSidebar : WidgetLocation.RightSidebar;
```

toxoxi commented on webpack.config.js:
```diff
@@ -111,6 +111,7 @@ if (isProd) {
     watchFiles: ['src/*'],
     headers: {
       'Access-Control-Allow-Origin': '*',
+      "Access-Control-Allow-Headers": "baggage, sentry-trace"
```
For what reason do we need? 👀

sotiwin replied:
It was needed for dev server to work when using with remnote opened in a browser. See in the remnote template: https://github.com/remnoteio/remnote-plugin-template-react/commit/a9c1590787e6a4ea0bb5fb4610da4547edea9866

toxoxi replied:
I got it! I suppose remnote uses Sentry to track bugs and those headers are required to do it.

toxoxi commented on package.json:
```diff
@@ -37,7 +37,7 @@
-    "webpack-cli": "^4.10.0",
-    "webpack-dev-server": "^4.9.3"
+    "webpack-cli": "^5.1.4",
+    "webpack-dev-server": "^5.0.4"
```
Thanks for upgrading 👍

toxoxi commented on public/manifest.json:
```diff
@@ -1,21 +1,22 @@
-  "id": "plant-znddugq",
+  "enableOnMobile": true,
+  "id": "sotiwin-3pl7w9y",
   "manifestVersion": 1,
-  "repoUrl": "https://github.com/toxoxi/remnote-table-of-contents",
+  "repoUrl": "https://github.com/sotiwin/remnote-table-of-contents",
   "version": {
     "major": 1,
-    "minor": 0,
-    "patch": 2
+    "minor": 1,
+    "patch": 0
   },
   "requestNative": false,
   "requiredScopes": [
     {
       "type": "All",
-      "level": "ReadCreateModify"
+      "level": "Read"
     }
-  ]
+  ],
+  "unlisted": true
```
Would some changes be reverted before merge? (author/id/repoUrl/unlisted)

toxoxi reviewed (commented):
I'm sorry for late review. I left some comments.

toxoxi commented:
@sotiwin 
> What do you think of using the left sidebar? Maybe there are other options. For example put an icon in the top of the page, which would open table of contents in a floating window.

I guess the current approach is good because:
- it delivers value with fewer changes
- it naturally fits with the native feature (sidebar) of RemNote

However, a floating window could be great if we can develop it well. Feel free to try it if you're interested :)

sotiwin committed:
use const for os and location

sotiwin committed:
revert manifest.json changes

sotiwin commented:
> @sotiwin
> 
> > What do you think of using the left sidebar? Maybe there are other options. For example put an icon in the top of the page, which would open table of contents in a floating window.
> 
> I guess the current approach is good because:
> 
> * it delivers value with fewer changes
> * it naturally fits with the native feature (sidebar) of RemNote
> 
> However, a floating window could be great if we can develop it well. Feel free to try it if you're interested :)

Okay, maybe in the future :)
For now I updated the merge request and reverted the manifest changes.

toxoxi commented:
hmm, I couldn't check its behavior because of this error.
![image](https://github.com/user-attachments/assets/5ce6eab4-c924-4d01-832e-6baed612e86a)

However, the screenshot looks good so I approve this. Thank you your contribution!

toxoxi commented:
I'll merge it and upload to plugin store

toxoxi commented:
> hmm, I couldn't check its behavior because of this error.

I had totally forgotten the tricky nature that it doesn't work with browser but with the native app.

Great work!

![image](https://github.com/user-attachments/assets/5b536964-75bc-4856-b0a9-fc6d3eb5a46c)
